handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -303,8 +303,24 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     private void replaceHandler(ChannelHandlerContext ctx, Selection selection) {
-        this.selection = selection;
-        SslHandler sslHandler = selection.context.newHandler(ctx.alloc());
+        try {
+            this.selection = selection;
+            replaceHandler(ctx, selection.hostname, selection.context);
+        } catch (Throwable cause) {
+            this.selection = EMPTY_SELECTION;
+            ctx.fireExceptionCaught(cause);
+        }
+    }
+
+    /**
+     * Replaces the {@link SniHandler} with a {@link SslHandler} as provided by the
+     * given {@link SslContext}.
+     *
+     * Users may override this method to implement custom behavior and to get a
+     * hold onto the {@link SslContext}.
+     */
+    protected void replaceHandler(ChannelHandlerContext ctx, String hostname, SslContext context) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "hostname". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)